### PR TITLE
[GSoC 2017] Porting Swift to Haiku - Final Report

### PR DESCRIPTION
**Pull request by return**

### CODE REVIEW

File: content/blog/return0e/2017-08-28_[gsoc_2017]_porting_swift_to_haiku_-_final_report.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+* Support for integrated REPL (non-LLDB version) works
+* Initial port of corelibs-foundation and libdispatch.
+* Run tests against the newly built standard library and swiftc.
+* Swift 3.1 recipe available on HaikuPorts.
+ 
+## What needs to be done (After GSoC):
+* Package Manager support via [llbuild](https://github.com/apple/swift-llbuild).
+* Debugging Swift via LLDB.
+* Further testing support.
+* Building SourceKit for Haiku.
```
Comment (korli) [inline content/blog/return0e/2017-08-28_[gsoc_2017]_porting_swift_to_haiku_-_final_report.md]: could you add a link to lldb and SourceKit?

File: content/blog/return0e/2017-08-28_[gsoc_2017]_porting_swift_to_haiku_-_final_report.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+Here's a short review below on what has been completed and still outstanding on this project and the current [documentation](https://gist.github.com/return/c9e28df2ea0009310835e51996a379c9)  to get Swift building on Haiku.
+
+## What has been completed:
+* Building swiftc and its standard library.
+* Execute simple swift programs (compiled or interpreted).
+* Support for integrated REPL (non-LLDB version) works
+* Initial port of corelibs-foundation and libdispatch.
+* Run tests against the newly built standard library and swiftc.
+* Swift 3.1 recipe available on HaikuPorts.
```
Comment (korli) [inline content/blog/return0e/2017-08-28_[gsoc_2017]_porting_swift_to_haiku_-_final_report.md]: A list URLs pointing to these completed items would be nice, ie https://github.com/return/swift-corelibs-libdispatch/commits/swift-3.1-haiku

Comment (return) [reply content/blog/return0e/2017-08-28_[gsoc_2017]_porting_swift_to_haiku_-_final_report.md]: @korli For some completed tasks, could a link to the blog post that mentions this task e.g [Integrated REPL support](https://www.haiku-os.org/blog/return0e/2017-07-28_gsoc_2017_porting_swift_to_haiku_-_week_8/) be acceptable?

Comment (korli) [reply content/blog/return0e/2017-08-28_[gsoc_2017]_porting_swift_to_haiku_-_final_report.md]: For me who knows a bit the project it's already difficult to follow and find the parts you worked on. Blog posts are interesting but don't help to summarize where patches you produced reside.
Keep in mind that's also a deliverable for GSoC (probably in a zip file though).

Comment (return) [reply content/blog/return0e/2017-08-28_[gsoc_2017]_porting_swift_to_haiku_-_final_report.md]: @korli I'm very sorry about the lack of information on my progress. I've recently documented my progress in a [github gist](https://gist.github.com/return/c9e28df2ea0009310835e51996a379c9) if that helps to clear up the confusion.

As for the deliverables, most of my work is in the upstream [PR](https://github.com/apple/swift/pull/11583),  So I think that creating a zip that has a script that re-creates the recent build environment with the documentation should acceptable for work submission.

Comment (korli) [reply content/blog/return0e/2017-08-28_[gsoc_2017]_porting_swift_to_haiku_-_final_report.md]: maybe just add the content of your github gist in the swift haikuports PR.

**Pull request merged**